reframe/frontend/cli.py

```diff
@@ -236,6 +236,9 @@ def main():
     misc_options.add_argument(
         '--non-default-craype', action='store_true', default=False,
         help='Test a non-default Cray PE')
+    misc_options.add_argument(
+        '--rerun-failed', action='store_true',
+        help='Print run options for all failed tests in summary table')
```

Comment:
The description of this option and its name are a bit confusing. It can be confused with the `--max-retries` options, which actually reruns the failed tests. Also, I'm not sure about the purpose of this option, because you get again an additional line for each test that failed, which is essentially the same information provided in the detailed report just above. You could simply add a line in the test failure info section, such as "Rerun as: ..."

Reply:
@vkarak  Thank you. I think It is a good idea to separate the statistics from failure_report and I can add  an option as '--failure-stats' for control.

My original report script parses the section of "SUMMARY OF FAILURES" and generates a shell script to rerun failed tests. The re-run script can run repeatedly until all tests are passed (it creates a working directory then a 'ok' file is touched in the working directory if the test is passed). It can help me to remove those "false positive" failures which are killed by system or due to unstable numerical reference. When I transfer this tool to ReFrame code, I try to keep this function. Anyway, "Rerun as:" is definitely useful for me to recover this re-run function. 

I will test changes in my local repos first. 

-ZQ

Reply:
@ZQyou Have you tried the `--max-retries=N` option of ReFrame. It will retry each failed test case N times automatically before it reports it as a failure.

Reply:
@vkarak  I do set `--max-retries=1` in my test but I don't need to rerun/fix some failures which are "no module" cases ("show module" cases below). I have a big set of tests running on three systems. Not all systems have  identical software environment. I don't want to check one by one so I decide to let them failed and discard them when I do re-run:
```
FAILURE STATISTICS

Total number of tests: 380
Total number of failures: 128

Phase         #     Failing tests
------------- ----- ----------------------------------------------------------
run           118   mkl_default_gnu|hdf5_default|hdf5_serial_default

 show module        bam2fastq_version|bam2fastq_version|bamtools_test|
                    bamtools_test|bamtools_version|bamtools_version|
                    .....
 load module        caffe_version|caffe_version|espresso_scf_mpi_test|
                    ....
```
I like the feature `--rerun-failed` in CTest which automatically pick up the failures and rerun them from the output directory. It allows me to fix the issues "later" and saves me time from bookkeeping.  Is there any possibility to implement such feature in ReFrame?

Reply:
> Not all systems have identical software environment. I don't want to check one by one so I decide to let them failed and discard them when I do re-run:

So essentially  you want to ignore any failure happening in the setup  phase?

> I like the feature --rerun-failed in CTest which automatically pick up the failures and rerun them from the output directory.

What is the difference to `--max-retries`?

Reply:
> So essentially you want to ignore any failure happening in the setup phase?
Yes. Some packages installed on A system can be deployed on new system soon. Basically I skip those failures with "show module" in reasons.

Reply:
> What is the difference to `--max-retries`?

`--rerun-failed` in CTest allows you to re-run failures any time as long as the configuration is not changed in the working directory. For example,  I run 10 tests and one of them is failed. I go to fix the source code of that test. Then  come back to the working folder and run `ctest --rerun-failed`.  I don't think `--max-retries` is for post re-run?

Reply:
Now, I see. `--max-retries` is only for the current run. That would be a nice feature but for that we need first to implement a mechanism to dump the output of the run, for example in a JSON file. It's easy to do, since all the information is available. I will open an issue for that. So for the moment I would say to keep this PR just for the condensed statistics report and make your `--rerun-failed` later.